validator-auction-rest-server/lib/config.js

```diff
@@ -0,0 +1,39 @@
+import { resolve } from 'path'
+import packageJson from '../package.json'
+
+require('dotenv')
+
+const config = {
+    env: process.env.NODE_ENV || 'production',
+
+    name: process.env.APPLICATION_NAME || packageJson.name,
+    version: packageJson.version,
+    description: packageJson.description,
+
+    path: {
+        logs: process.env.PATH_LOGS || resolve(__dirname, '../logs')
+    },
+
+    api: {
+        server: {
+            host: process.env.API_SERVER_HOST || '127.0.0.1',
+            port: process.env.API_SERVER_PORT || 8090,
+            basePath: process.env.API_SERVER_BASEPATH || ''
+        }
+    },
+    validatorAuction: {
+        network: process.env.VALIDATOR_NETWORK || undefined,
+        contractAddress: process.env.VALIDATOR_ADDRESS || undefined
+    },
+    database: {
+        ethEvents: {
+            host: process.env.ETH_EVENTS_URL || undefined,
+            token: process.env.ETH_EVENTS_TOKEN || undefined,
+
+            defaultSize: 10,
+            maxSize: 5000
+        }
+    }
+}
+
```

Comment:
I do think most of our code will not work if we hit the '|| undefined' branch above. Can we make the program abort in that case with a nice error message?  Also, I do think there `ETH_EVENTS_URL` could use `https://api.eth.events` as default.

Reply:
I have added some consistency checks during the server startup to enforce the documented environment variables. To that effect, I also removed the explicit `undefined` default values.

ETH_EVENTS_URL can be undefined to start the backend-server in test mode.